scipy/stats/_hypotests.py

```diff
@@ -699,12 +699,11 @@ def barnard_exact(table, alternative="two-sided", pooled=True, n_iter=1):
         t-test) or unpooled variance (Welch's t-test). Default is ``True`` :
         Statistic test is computed using pooled variance.
 
-    n_iter : int, optional
-        Number of iterations in shgo's alrogithm optimum search. Default is 1.
-        Must be non-negative. In most cases, 3 iterations is perfectly
-        enough to reach good precision. Above a certain number (around 6
-        iterations), the result will not change anymore. Note that every
-        iteration added comes with a performance cost.
+    shgo_n : int, optional
+        Number of sampling points used in the construction of the sobol
```

Comment:
Or alternatively the method so a reference to ```  `scipy.stats.qmc.Sobol` ```
```suggestion
        Number of sampling points used in the construction of the Sobol'
```